### PR TITLE
ccs-2989 fix and pagination changes

### PR DESCRIPTION
**Pull request by ankkit24**
Fixed the issue of refresh to the first page on a new search.
Also, added a custom pagination component.

### CODE REVIEW

File: frontend/src/app/Pagination.tsx [hunk 1]
```diff
@@ -1,65 +1,128 @@
...
+    isOpen: false,
+    itemsPerPage: 25,
+    renderSearch: false,
+    searchValue: '',
+    selected: this.dropdownItems[0]
+
+  };
+
+  public button1 = true;
+  public button2 = true;
```
Comment (jmarrero) [inline frontend/src/app/Pagination.tsx]: These button1, button2, button3 should be renamed to something more significant.

File: frontend/src/app/Pagination.tsx [hunk 2]
```diff
@@ -1,65 +1,128 @@
...
+                            <button disabled={this.button2} data-action="previous" aria-label="Go to previous page" className="pf-c-button pf-m-plain" type="button" onClick={this.props.handleMoveLeft}><svg fill="currentColor" height="1em" width="1em" viewBox="0 0 256 512" aria-hidden="true" role="img">
+                              <path d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z" transform="" /></svg></button>
+                            Page: {this.props.pageNumber}
+                            <button disabled={this.button3} data-action="next" aria-label="Go to next page" className="pf-c-button pf-m-plain" type="button" onClick={this.props.handleMoveRight}><svg fill="currentColor" height="1em" width="1em" viewBox="0 0 256 512" aria-hidden="true" role="img">
+                              <path d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z" transform="" /></svg></button>
```
Comment (jmarrero) [inline frontend/src/app/Pagination.tsx]: I see these
```
<path d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z" transform="" /></svg></button>
```
All over the place, I am not sure what that is accomplishing. Can you explain why we need that path and why it has that structure?

File: frontend/src/app/search.tsx [hunk 3]
```diff
@@ -32,7 +32,8 @@ export default class Search extends Component {
     pageLimit: 10,
     redirect: false,
     redirectLocation: '',
-    sortKey: ''
+    showOptions: true,
```
Comment (jmarrero) [inline frontend/src/app/search.tsx]: showOptions? options related to what? maybe we need to rename this also.

Comment (ankkit24) [reply frontend/src/app/search.tsx]: Thanks for pointing this out Joseph.
The drop down options are hidden at the bottom of the page, so had to do this check.
I could change the name to 'showDropdownOptions' instead. Does that variable name sound good?

File: frontend/src/app/Pagination.tsx (hunk 2 above)
Comment (jmarrero) [reply frontend/src/app/Pagination.tsx]: Nice thanks

File: frontend/src/app/search.tsx (hunk 3 above)
Comment (jmarrero) [reply frontend/src/app/search.tsx]: yes

File: frontend/src/app/Pagination.tsx (hunk 1 above)
Comment (ankkit24) [reply frontend/src/app/Pagination.tsx]: Yes Joseph!
I am changing button1 to firstPageButton, button2 to previousPageButton and button3 to nextPageButton as per their functionality

Comment (jmarrero) [reply frontend/src/app/Pagination.tsx]: sounds good.

**Comment by jmarrero** [comment]
UI rework looks nice
Code LGTM
Coverage went up
Functions still work as expected.

**Pull request merged**